streams.go

```diff
@@ -366,3 +363,46 @@ func (r *ByteStreamReader) ReadAll() []byte {
 type TextStreamHandler func(reader *TextStreamReader, participantIdentity string)
 
 type ByteStreamHandler func(reader *ByteStreamReader, participantIdentity string)
+
+// ---------------------------------------------------------
+
+func chunkUtf8String(s string) [][]byte {
```

Comment:
would be good to have some UT and also some comments as to what this is doing would be great.